File: src/data/courses/dsa/two-pointers-sliding-window/sliding-window-fixed-size/find-all-anagrams/code/cpp/find_all_anagrams_sol1.cpp
class Solution {
public:
    vector<int> findAnagrams(string s, string p) {
        vector<int> result;
        int m = p.size();
        vector<int> pFreq(26, 0);
        for (char c : p) pFreq[c-'a']++;
        for (int i = 0; i <= (int)s.size()-m; i++) {
            vector<int> wFreq(26, 0);
            for (int j = i; j < i+m; j++) wFreq[s[j]-'a']++;
            if (wFreq == pFreq) result.push_back(i);
        }
        return result;
    }
};

File: src/data/courses/dsa/two-pointers-sliding-window/sliding-window-fixed-size/find-all-anagrams/code/python/find_all_anagrams_sol1.py
class Solution:
    def findAnagrams(self, s, p):
        from collections import Counter
        result, m = [], len(p)
        pf = Counter(p)
        for i in range(len(s)-m+1):
            if Counter(s[i:i+m]) == pf: result.append(i)
        return result